src/components/dashboard/components/dictionary/DictionaryContainer.jsx

```diff
@@ -76,9 +81,41 @@ export class DictionaryOverview extends Component {
     const url = headVersionObj.version_url;
     this.props.releaseHead(url, data);
   }
+
   handleHide = () => this.setState({ showEditModal: false });
   handleShow = () => this.setState({ showEditModal: true });
 
+  hideVersionModal = () => this.setState({ openVersionModal: false });
+  showVersionModal = () => this.setState({ openVersionModal: true });
+
+  handleChange = e => {
+    const { name, value } = e.target;
+    this.setState({ [name]: value });
+  }
+
+  handleCreateVersion = event => {
+    event.preventDefault();
+    const {
+      match: {
+        params: {
+          ownerType, owner, type, name,
+        },
+      },
+    } = this.props;
+    const url = `/${ownerType}/${owner}/${type}/${name}/versions/`;
+    const data = {
+      id: this.state.versionId,
+      released: true,
+      description: this.state.versionDescription,
+    }
+    this.props.createVersion(url, data)
+    setTimeout(() => {
```

Comment:
Why have you added a delay of 2 seconds?

Reply:
To determine whether the call to the API was sucessfull so as to close the modal. if not the modal should remain open.

Reply:
That seems like a bad assumption. You are not in charge of what happens over the network. Plus you could overestimate or underestimate. Why don't you handle asynchronous actions the redux way?

Reply:
That seems like a bad assumption. You are not in charge of what happens over the network. Plus you could overestimate or underestimate. Why don't you handle asynchronous actions the redux way?

Reply:
That is a very good way of implementing it. Will actually do that.